frontend/src/components/PasswordChange.jsx

```diff
@@ -0,0 +1,378 @@
+                            : ""
+                        }
+                        type={showConfirmPassword ? "text" : "password"}
+                        placeholder="Type your password again"
+                        onChange={(e) => {
+                          setConfirmPassword(e.target.value);
+                          checkPasswordsMatch(password, e.target.value);
+                          if (e.target.value.length === 0) {
+                            setConfirmPasswordValidated(false);
+                          }
+                        }}
+                        value={confirmPassword}
+                      />
+                      <InputGroup.Text
+                        className="password-icon"
+                        onClick={() =>
+                          setShowConfirmPassword(!showConfirmPassword)
+                        }
+                      >
+                        {showConfirmPassword ? <EyeSlash /> : <Eye />}
+                      </InputGroup.Text>
+                      <Form.Control.Feedback type="invalid">
+                        {confirmPasswordErrorMessage}
+                      </Form.Control.Feedback>
+                    </InputGroup>
+                  </Form.Group>
+                  <Row>
+                    <Col sm={3}></Col>
+                    <Col sm={6}>
+                      <Button
+                        className="w-100 mb-4 border-0"
+                        variant="primary"
+                        type="submit"
+                        id="formSubmitButton"
+                      >
+                        Set Password
+                      </Button>
+                    </Col>
+                    <Col sm={3}></Col>
+                  </Row>
+                </Form>
+              </Col>
+            </Row>
+          </Col>
+          <Col sm={4}></Col>
+        </Row>
+      </Container>
+      <Modal
+        show={showUpdateConfirmation}
+        size="sm"
+        onHide={() => navigate("/login")}
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>
+            {updateSuccess
+              ? "Password update success"
+              : "Password update Error"}
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Row>
+            <Col className="text-center mb-4">
+              {updateSuccess ? (
+                <CheckCircleFill className="text-success" size={100} />
+              ) : (
+                <ExclamationTriangleFill className="text-warning" size={100} />
+              )}
+            </Col>
+          </Row>
+          <Row>
+            <Col className="text-center">
+              <p>{updateConfirmationMessage}</p>
+            </Col>
+          </Row>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            variant={updateSuccess ? "success" : "secondary"}
+            onClick={() => navigate("/login")}
+          >
+            {updateSuccess ? "Go to login" : "Close"}
```

Comment:
Fulfills the missing aceptance criteria of updating the password and redirecting to the login.